File: test_extreme/javascript/medium_ai.js
// JS 中型 AI代码 — 安全违规 + 高CC + 空catch + 重复
const DB_PASSWORD = "js_db_pass"; // [BLOCK]
const SECRET = "secret123";       // [BLOCK]

function processOrder(order, user, items, discountCode, shipping, express, gift, loyalty) {
    let total = 0;
    for (const item of items) total += item.price * item.qty;
    if (discountCode) {
        if (discountCode === "SAVE10") total *= 0.9;
        else if (discountCode === "SAVE20") total *= 0.8;
        else { if (loyalty === "gold") { total *= 0.95; if (express) { total += 25; if (gift) { total += 10; } } } }
    }
    if (shipping === "overnight") total += 45;
    else if (shipping === "express") total += 25;
    return total;
} // CC≈10 → WARN

// 重复查询函数
function getUser(n) { return fetch("/api/users?name=" + n); }
function getOrder(n) { return fetch("/api/orders?name=" + n); }
function getProduct(n) { return fetch("/api/products?name=" + n); }

// 空catch
function risky(url) { try { fetch(url); } catch(e) {} }

// [BLOCK] 日志泄露
console.debug("DB password: " + DB_PASSWORD);

File: test_extreme/go/medium_ai.go
// Go 中型 AI代码 — 安全违规 + 高CC + 空错误处理
package main

import "os"

var DB_PASSWORD = "go_pass_123"  // [BLOCK]
var SECRET_KEY = "go_secret_456"  // [BLOCK]

func ProcessOrder(items []float64, tier string, holiday bool, points int,
    region string, currency string, gift bool, promo string,
    months int, express bool) float64 {  // 10 params → BLOCK
    total := 0.0
    for _, item := range items { total += item }
    if tier == "gold" { total *= 0.8
    } else if tier == "silver" { total *= 0.9
    } else { if holiday { total *= 0.95
        if points > 1000 { total -= 100 } } }
    if region == "EU" { total *= 1.21
        if currency == "GBP" { total *= 0.85
            if gift { total *= 0.9
                if promo != "" { total *= 0.85 } } } }
    if months > 12 { total *= 1.05
    } else if months > 6 { total *= 1.03
    } else if months > 3 { total *= 1.01 }
    return total
} // CC≈13 → WARN

func riskyCmd(cmd string) {
    // [BLOCK] os/exec
    os.Setenv("CMD", cmd)
}

func main() {}
